File: algorithms/2180.count-integers-with-even-digit-sum/solution.rb
# @param {Integer} num
# @return {Integer}
def count_even(num)
  res = 0

  (1..num).each do |i|
    sum = i.to_s.chars.map(&:to_i).sum
    if sum % 2 == 0
      res += 1
    end
  end

  res
end

File: algorithms/2180.count-integers-with-even-digit-sum/solution.go
func countEven(num int) int {
	res := 0

	for i := 1; i <= num; i++ {
		if digitSum(i)%2 == 0 {
			res++
		}
	}

	return res
}

func digitSum(num int) int {
	sum := 0
	for _, r := range strconv.Itoa(num) {
		sum += int(r - '0')
	}

	return sum
}
